src/Collapse.js

```diff
@@ -34,7 +35,12 @@ const Collapse = React.createClass({
 
 
   componentWillReceiveProps({isOpened}) {
-    this.setState({isOpenedChanged: isOpened !== this.props.isOpened});
+    var isOpenedChanged = isOpened !== this.props.isOpened;
+    this.setState({isOpenedChanged}, () => {
+      if (this.props.onHeightReady && isOpenedChanged) {
+        this.props.onHeightReady(isOpened ? this.state.height : 0);
```

Comment:
I'll look into using `componentDidUpdate` instead. Shouldn't really be a problem